flang/runtime/io-stmt.cpp

```diff
@@ -119,6 +120,9 @@ template <Direction DIR> void InternalIoStatementState<DIR>::BackspaceRecord() {
 }
 
 template <Direction DIR> int InternalIoStatementState<DIR>::EndIoStatement() {
+  if constexpr (DIR == Direction::Output) {
```

Comment:
If it is possible for the buffer flush to raise an error that the program can catch via `ERR=` or `IOSTAT=`, do this in `CompleteOperation` instead.